# MVP — Engenharia de Dados

**Nome:** Guilherme Montenegro Banharo  
**Matrícula:** 16660054790  
**Data:** 05/09/2026  
**Dataset:** Framingham Heart Disease Dataset  
**Fonte:** [Kaggle — Logistic Regression To Predict Heart Disease](https://www.kaggle.com/datasets/dileep070/heart-disease-prediction-using-logistic-regression)




## Pipeline de Dados

Este notebook implementa um pipeline de dados em ambiente de nuvem utilizando o Databricks.

O pipeline será organizado seguindo a Arquitetura Medalhão:

- **Bronze:** armazenamento dos dados brutos;
- **Silver:** limpeza, tratamento e padronização dos dados;
- **Gold:** dados tratados e organizados para consumo analítico e resposta às perguntas de negócio.

# 1. Definição do problema

## 1.1 Descrição do problema

As doenças cardiovasculares estão entre os principais problemas de saúde pública e sua ocorrência pode estar relacionada a diferentes características demográficas, comportamentais e clínicas dos indivíduos.

Neste projeto será utilizado o conjunto de dados Framingham Heart Disease, contendo informações como idade, tabagismo, pressão arterial, colesterol, diabetes, índice de massa corporal e glicose, além da variável `TenYearCHD`, que indica a ocorrência ou não de doença coronariana em um período de dez anos.
A partir desses dados, busca-se compreender como determinadas características dos pacientes se relacionam com a ocorrência de doença coronariana, transformando os dados brutos em informações organizadas e adequadas para análise.

## 1.2 Objetivo do MVP

O objetivo deste MVP é construir um pipeline de dados ponta a ponta em ambiente de nuvem, responsável pela ingestão, armazenamento, tratamento, organização e disponibilização dos dados do estudo de Framingham para análise.

Ao final do pipeline, espera-se obter dados tratados e estruturados que permitam analisar características demográficas, comportamentais e clínicas e responder às perguntas definidas para o projeto.

## 1.3 Perguntas de negócio

Para orientar a construção do pipeline e a análise dos dados, foram definidas as seguintes perguntas:

1. Como a ocorrência de doença coronariana em dez anos varia de acordo com a faixa etária dos pacientes?
2. Pacientes com hipertensão apresentam maior ocorrência de doença coronariana em dez anos?
3. Como indicadores clínicos como colesterol, pressão arterial, IMC e glicose diferem entre pacientes com e sem ocorrência de doença coronariana?
4. Como fatores relacionados ao tabagismo se relacionam com a ocorrência de doença coronariana em dez anos?
5. Pacientes com diabetes apresentam maior ocorrência de doença coronariana em dez anos quando comparados aos pacientes sem diabetes?

Essas perguntas servirão como direcionadoras das etapas seguintes do projeto, orientando quais dados serão utilizados, quais transformações serão realizadas e como os dados serão organizados para a análise final.

# 2. Seleção dos Dados

## 2.1 Fonte dos dados

Para o desenvolvimento deste MVP será utilizado o Framingham Heart Disease Dataset, disponibilizado publicamente na plataforma Kaggle. O conjunto reúne informações demográficas, comportamentais e clínicas relacionadas à saúde cardiovascular de participantes do estudo de Framingham, Massachusetts.

A base contém informações relacionadas a diferentes fatores de risco cardiovascular e inclui a variável TenYearCHD, que representa a ocorrência ou não de doença coronariana em um período de dez anos. Esses dados serão utilizados para apoiar as análises relacionadas às perguntas de negócio definidas neste projeto.

**Fonte dos dados:** Kaggle — Logistic Regression To Predict Heart Disease

**Link:** https://www.kaggle.com/datasets/dileep070/heart-disease-prediction-using-logistic-regression

## 2.2 Justificativa da escolha

A base foi escolhida por estar alinhada ao problema definido para o projeto e por reunir diferentes características que podem ser analisadas em relação à ocorrência de doença coronariana.

Além disso, o conjunto apresenta características relevantes para a construção de um pipeline de Engenharia de Dados, como a presença de valores ausentes, atributos com diferentes significados e domínios e a necessidade de verificações de consistência e qualidade.

Essas características permitem aplicar, de forma prática, as etapas de ingestão, armazenamento, tratamento, transformação, organização e disponibilização dos dados para análise.

A variedade de informações disponíveis também possibilita a construção de diferentes visões dos dados para responder às perguntas de negócio estabelecidas no início do projeto.

## 2.3 Licença e condições de uso

O conjunto de dados encontra-se publicamente disponível na plataforma Kaggle. Entretanto, na página utilizada como fonte para este projeto, o campo referente à licença está identificado como "Unknown", não havendo uma licença formal explicitamente informada para essa publicação do conjunto de dados.

Essa condição será considerada uma limitação da fonte e será registrada na documentação do projeto. A disponibilidade pública dos dados no Kaggle não será interpretada como indicação de uma licença específica ou de autorização irrestrita para redistribuição ou utilização comercial.

Neste MVP, a base será utilizada exclusivamente no contexto acadêmico da disciplina, mantendo documentadas tanto a origem dos dados quanto a ausência de uma licença explicitamente informada na página consultada. Essa condição é considerada uma limitação da fonte utilizada.

# 3. Ingestão e Camada Bronze

## 3.1 Ingestão dos dados

A primeira etapa do pipeline consiste na ingestão do conjunto de dados original e seu armazenamento na camada Bronze.

Seguindo a Arquitetura Medalhão, a camada Bronze tem como objetivo preservar os dados em seu estado bruto, mantendo uma representação fiel da fonte antes da aplicação de regras de limpeza, tratamento ou transformação.

Nesta etapa serão realizadas apenas verificações iniciais necessárias para confirmar que os dados foram carregados corretamente.

In [0]:
%sql
CREATE SCHEMA IF NOT EXISTS workspace.framingham_mvp;

In [0]:
%sql
SHOW SCHEMAS IN workspace;

databaseName
default
framingham_mvp
information_schema


In [0]:
# Caminho do arquivo bruto no Unity Catalog
raw_path = "/Volumes/workspace/framingham_mvp/raw/framingham.csv"

# Leitura do arquivo CSV com Spark
df_bronze = (
    spark.read
    .option("header", "true")
    .option("inferSchema", "true")
    .csv(raw_path)
)

display(df_bronze)

male,age,education,currentSmoker,cigsPerDay,BPMeds,prevalentStroke,prevalentHyp,diabetes,totChol,sysBP,diaBP,BMI,heartRate,glucose,TenYearCHD
1,39,4,0,0,0,0,0,0,195,106.0,70.0,26.97,80,77,0
0,46,2,0,0,0,0,0,0,250,121.0,81.0,28.73,95,76,0
1,48,1,1,20,0,0,0,0,245,127.5,80.0,25.34,75,70,0
0,61,3,1,30,0,0,1,0,225,150.0,95.0,28.58,65,103,1
0,46,3,1,23,0,0,0,0,285,130.0,84.0,23.1,85,85,0
0,43,2,0,0,0,0,1,0,228,180.0,110.0,30.3,77,99,0
0,63,1,0,0,0,0,0,0,205,138.0,71.0,33.11,60,85,1
0,45,2,1,20,0,0,0,0,313,100.0,71.0,21.68,79,78,0
1,52,1,0,0,0,0,1,0,260,141.5,89.0,26.36,76,79,0
1,43,1,1,30,0,0,1,0,225,162.0,107.0,23.61,93,88,0


### 3.2 Validação inicial da ingestão

Após a leitura do arquivo bruto, foram realizadas verificações iniciais para confirmar a integridade da ingestão antes de sua persistência na camada Bronze.

Nesta etapa são verificadas a quantidade de registros, a quantidade de atributos e o esquema inferido pelo Spark. A inspeção do esquema permite identificar os tipos de dados atribuídos às variáveis durante a ingestão, sem realizar correções ou transformações neste momento. Eventuais ajustes de tipos e tratamento de valores ausentes serão realizados posteriormente na camada Silver.

In [0]:
# Validação inicial da ingestão
qtd_registros = df_bronze.count()
qtd_atributos = len(df_bronze.columns)

print(f"Quantidade de registros: {qtd_registros}")
print(f"Quantidade de atributos: {qtd_atributos}")

print("\nEsquema inferido pelo Spark:")
df_bronze.printSchema()

Quantidade de registros: 4238
Quantidade de atributos: 16

Esquema inferido pelo Spark:
root
 |-- male: integer (nullable = true)
 |-- age: integer (nullable = true)
 |-- education: string (nullable = true)
 |-- currentSmoker: integer (nullable = true)
 |-- cigsPerDay: string (nullable = true)
 |-- BPMeds: string (nullable = true)
 |-- prevalentStroke: integer (nullable = true)
 |-- prevalentHyp: integer (nullable = true)
 |-- diabetes: integer (nullable = true)
 |-- totChol: string (nullable = true)
 |-- sysBP: double (nullable = true)
 |-- diaBP: double (nullable = true)
 |-- BMI: string (nullable = true)
 |-- heartRate: string (nullable = true)
 |-- glucose: string (nullable = true)
 |-- TenYearCHD: integer (nullable = true)



### 3.3 Persistência da camada Bronze

Após a validação da ingestão, os dados são persistidos na camada Bronze em formato Delta. Essa camada mantém os dados conforme recebidos da fonte, sem aplicação de regras de limpeza ou transformação, servindo como ponto de partida para as etapas posteriores do pipeline.

A tabela Bronze será armazenada no Unity Catalog, dentro do catálogo `workspace` e do schema `framingham_mvp`, permitindo seu gerenciamento e acesso nas etapas seguintes.

In [0]:
# Persistência dos dados na camada Bronze
tabela_bronze = "workspace.framingham_mvp.bronze_framingham"

(
    df_bronze.write
    .format("delta")
    .mode("overwrite")
    .saveAsTable(tabela_bronze)
)

print(f"Tabela Bronze criada: {tabela_bronze}")

Tabela Bronze criada: workspace.framingham_mvp.bronze_framingham


Na camada Bronze, o arquivo original foi ingerido com Spark e posteriormente persistido como uma tabela Delta no Unity Catalog. Nessa etapa, os dados foram mantidos sem regras de limpeza ou transformação, preservando a representação recebida da fonte. Essa tabela servirá como origem para os tratamentos realizados na camada Silver.


### 3.4 Validação da camada Bronze

Após a persistência, a tabela Bronze é carregada novamente a partir do Unity Catalog para verificar se os dados foram armazenados corretamente.

Nesta validação são conferidas a quantidade de registros e a quantidade de atributos da tabela persistida, comparando-as com os valores identificados durante a ingestão.

In [0]:
# Leitura da tabela Bronze persistida
df_bronze_validacao = spark.table("workspace.framingham_mvp.bronze_framingham")

# Validação da quantidade de registros e atributos
registros_bronze = df_bronze_validacao.count()
atributos_bronze = len(df_bronze_validacao.columns)

print(f"Registros na tabela Bronze: {registros_bronze}")
print(f"Atributos na tabela Bronze: {atributos_bronze}")

Registros na tabela Bronze: 4238
Atributos na tabela Bronze: 16


A validação da tabela Bronze confirmou a presença de 4.238 registros e 16 atributos, correspondentes às mesmas quantidades identificadas durante a leitura do arquivo original. Dessa forma, verifica-se que a persistência dos dados na camada Bronze foi realizada corretamente, mantendo a estrutura e a quantidade de registros da fonte antes das etapas de limpeza e transformação.

## 4. Tratamento e Camada Silver

A camada Silver é responsável pela limpeza, padronização e preparação dos dados provenientes da camada Bronze.

Antes da aplicação das regras de tratamento, será realizada uma análise da qualidade dos dados para identificar valores ausentes, registros duplicados, tipos de dados inadequados e possíveis inconsistências. A partir desse diagnóstico serão definidas as transformações necessárias, buscando preservar a maior quantidade possível de informações relevantes para as análises e perguntas de negócio do projeto.

### 4.1 Diagnóstico da qualidade dos dados

Inicialmente, a tabela Bronze é utilizada como fonte para avaliar a qualidade do conjunto de dados. Nesta etapa serão investigados valores ausentes, duplicidades e os tipos de dados das variáveis, sem realizar alterações nos registros.

### 4.1.1 Valores ausentes

Antes da aplicação das regras de tratamento da camada Silver, foi realizada uma análise dos valores ausentes presentes na tabela Bronze.

A verificação considera tanto valores nulos quanto registros representados pelo texto `NA`, pois a inspeção inicial mostrou que algumas ausências da fonte foram carregadas dessa forma. O objetivo é identificar quais atributos necessitam de tratamento e apoiar a definição das transformações que serão aplicadas na camada Silver.

In [0]:
from pyspark.sql import functions as F

# Leitura da camada Bronze
df_silver_base = spark.table("workspace.framingham_mvp.bronze_framingham")

# Contagem de valores ausentes (null ou NA)
expressoes_ausentes = []

for coluna in df_silver_base.columns:
    expressoes_ausentes.append(
        F.sum(
            F.when(
                F.col(coluna).isNull() |
                (F.upper(F.trim(F.col(coluna).cast("string"))) == "NA"),
                1
            ).otherwise(0)
        ).alias(coluna)
    )

df_ausentes = df_silver_base.select(expressoes_ausentes)

display(df_ausentes)

male,age,education,currentSmoker,cigsPerDay,BPMeds,prevalentStroke,prevalentHyp,diabetes,totChol,sysBP,diaBP,BMI,heartRate,glucose,TenYearCHD
0,0,105,0,29,53,0,0,0,50,0,0,19,1,388,0


A análise identificou valores ausentes em sete atributos do conjunto de dados. A variável `glucose` apresentou a maior quantidade de ausências, com 388 registros, seguida por `education` (105), `BPMeds` (53), `totChol` (50), `cigsPerDay` (29), `BMI` (19) e `heartRate` (1). Os demais atributos não apresentaram valores ausentes.

Também foi observado que parte das ausências está representada pelo texto `NA` nos dados ingeridos, contribuindo para que algumas variáveis originalmente numéricas tenham sido inferidas pelo Spark como `string`. Esses casos serão tratados posteriormente durante a construção da camada Silver.

### 4.1.2 Registros duplicados

Além dos valores ausentes, foi verificada a existência de registros duplicados na camada Bronze. Essa análise permite identificar possíveis repetições antes da aplicação das regras de tratamento e construção da camada Silver.

In [0]:
# Verificação de registros duplicados

# Quantidade total de registros
qtd_total = df_silver_base.count()

# Quantidade de registros distintos
qtd_distintos = df_silver_base.distinct().count()

# Diferença entre total e distintos
qtd_duplicados = qtd_total - qtd_distintos

print(f"Quantidade total de registros: {qtd_total}")
print(f"Quantidade de registros distintos: {qtd_distintos}")
print(f"Quantidade de registros duplicados: {qtd_duplicados}")

Quantidade total de registros: 4238
Quantidade de registros distintos: 4238
Quantidade de registros duplicados: 0


A verificação identificou 4.238 registros na tabela Bronze, sendo todos distintos. Portanto, não foram encontrados registros duplicados no conjunto de dados.

Esse resultado indica que não será necessária a remoção de duplicidades durante o tratamento da camada Silver, preservando-se todos os registros existentes em relação a esse critério de qualidade.

### 4.1.3 Tipos de dados e inconsistências

Além da análise de valores ausentes e duplicidades, foram verificados os tipos de dados atribuídos às variáveis durante a ingestão.

Essa verificação tem como objetivo identificar atributos cujo tipo inferido pelo Spark não corresponde à natureza esperada da informação, especialmente em colunas numéricas que contêm o valor textual `NA`. Essas inconsistências deverão ser corrigidas durante a etapa de tratamento e padronização da camada Silver.

In [0]:
# Verificação dos tipos de dados da camada Bronze

tipos_dados = [
    (campo.name, campo.dataType.simpleString())
    for campo in df_silver_base.schema.fields
]

df_tipos = spark.createDataFrame(
    tipos_dados,
    ["atributo", "tipo_inferido"]
)

display(df_tipos)

atributo,tipo_inferido
male,int
age,int
education,string
currentSmoker,int
cigsPerDay,string
BPMeds,string
prevalentStroke,int
prevalentHyp,int
diabetes,int
totChol,string


A inspeção dos tipos de dados mostrou que algumas variáveis de natureza numérica foram inferidas pelo Spark como `string`: `education`, `cigsPerDay`, `BPMeds`, `totChol`, `BMI`, `heartRate` e `glucose`.

Esse comportamento está relacionado à presença do valor textual `NA` no arquivo de origem. Como a camada Bronze tem o objetivo de preservar os dados conforme recebidos, nenhuma correção foi realizada durante a ingestão.

Na camada Silver, esses valores serão convertidos para nulos e os atributos serão padronizados para tipos compatíveis com sua natureza, permitindo a realização adequada das análises posteriores.

## 4.2 Tratamento e padronização dos dados

Com base nos problemas identificados durante a análise de qualidade, são aplicadas as transformações necessárias para a construção da camada Silver.

Nesta etapa, os valores textuais `NA` são convertidos para valores nulos e os atributos são convertidos para tipos de dados compatíveis com sua natureza. A tabela Bronze permanece inalterada, sendo as transformações realizadas em um novo DataFrame destinado à camada Silver.

### 4.2.1 Conversão de `NA` e padronização dos tipos

In [0]:
from pyspark.sql import functions as F

# Criação do DataFrame de trabalho da camada Silver
df_silver = df_silver_base

# Colunas que foram inferidas como string,
# mas representam informações numéricas
colunas_numericas_string = [
    "education",
    "cigsPerDay",
    "BPMeds",
    "totChol",
    "BMI",
    "heartRate",
    "glucose"
]

# Converte "NA" para null e padroniza as colunas como double
for coluna in colunas_numericas_string:
    df_silver = df_silver.withColumn(
        coluna,
        F.when(
            F.upper(F.trim(F.col(coluna))) == "NA",
            F.lit(None)
        )
        .otherwise(F.col(coluna))
        .cast("double")
    )

# Visualização do novo esquema
df_silver.printSchema()

root
 |-- male: integer (nullable = true)
 |-- age: integer (nullable = true)
 |-- education: double (nullable = true)
 |-- currentSmoker: integer (nullable = true)
 |-- cigsPerDay: double (nullable = true)
 |-- BPMeds: double (nullable = true)
 |-- prevalentStroke: integer (nullable = true)
 |-- prevalentHyp: integer (nullable = true)
 |-- diabetes: integer (nullable = true)
 |-- totChol: double (nullable = true)
 |-- sysBP: double (nullable = true)
 |-- diaBP: double (nullable = true)
 |-- BMI: double (nullable = true)
 |-- heartRate: double (nullable = true)
 |-- glucose: double (nullable = true)
 |-- TenYearCHD: integer (nullable = true)



A padronização dos tipos foi realizada com sucesso. Os valores textuais `NA` foram convertidos para valores nulos e as variáveis numéricas que haviam sido inferidas como `string` foram convertidas para o tipo `double`.

Com essa transformação, os atributos podem ser utilizados adequadamente em operações estatísticas e nas etapas posteriores de tratamento. A camada Bronze permanece inalterada, enquanto as modificações são aplicadas somente ao DataFrame de trabalho da camada Silver.

### 4.2.2 Análise para tratamento dos valores ausentes

Após a conversão dos valores `NA` para nulos, são analisadas as variáveis que apresentam dados ausentes. Essa etapa tem como objetivo definir uma estratégia de tratamento compatível com a natureza e a distribuição de cada atributo, evitando a aplicação indiscriminada de uma única regra de imputação.

In [0]:
# Variáveis com valores ausentes
colunas_com_ausencias = [
    "education",
    "cigsPerDay",
    "BPMeds",
    "totChol",
    "BMI",
    "heartRate",
    "glucose"
]

# Estatísticas descritivas
display(
    df_silver.select(colunas_com_ausencias).summary(
        "count",
        "mean",
        "stddev",
        "min",
        "25%",
        "50%",
        "75%",
        "max"
    )
)

summary,education,cigsPerDay,BPMeds,totChol,BMI,heartRate,glucose
count,4133,4209,4185,4188,4219,4237,3850
mean,1.9789499153157513,9.003088619624615,0.02962962962962963,236.72158548233045,25.80200758473571,75.87892376681614,81.96675324675324
stddev,1.0197906893121707,11.920093587827989,0.16958356790200144,44.59033431874948,4.080111062409023,12.026596351612573,23.95999818923739
min,1.0,0.0,0.0,107.0,15.54,44.0,40.0
25%,1.0,0.0,0.0,206.0,23.07,68.0,71.0
50%,2.0,0.0,0.0,234.0,25.4,75.0,78.0
75%,3.0,20.0,0.0,263.0,28.04,83.0,87.0
max,4.0,70.0,1.0,696.0,56.8,143.0,394.0


A análise das estatísticas descritivas foi utilizada para orientar a estratégia de tratamento dos valores ausentes. Para as variáveis numéricas contínuas (`cigsPerDay`, `totChol`, `BMI`, `heartRate` e `glucose`), será utilizada a mediana, por ser menos sensível à influência de valores extremos.

Para `education` e `BPMeds`, que representam informações discretas ou categóricas codificadas numericamente, será utilizada a moda. Dessa forma, o tratamento considera a natureza de cada atributo em vez de aplicar uma única regra de imputação a todo o conjunto de dados.

### 4.2.3 Tratamento dos valores ausentes

Com base na análise anterior, são aplicadas as estratégias de imputação definidas para cada grupo de variáveis. Os valores ausentes das variáveis numéricas contínuas são substituídos pela mediana de cada atributo, enquanto `education` e `BPMeds` são preenchidas pela respectiva moda.

As transformações são realizadas exclusivamente no DataFrame destinado à camada Silver, preservando os dados originais armazenados na camada Bronze.

In [0]:
from pyspark.sql import functions as F

# Variáveis numéricas contínuas
# Tratamento dos valores ausentes pela mediana
colunas_mediana = [
    "cigsPerDay",
    "totChol",
    "BMI",
    "heartRate",
    "glucose"
]

for coluna in colunas_mediana:
    mediana = df_silver.approxQuantile(
        coluna,
        [0.5],
        0.001
    )[0]

    df_silver = df_silver.fillna({
        coluna: mediana
    })

# Variáveis discretas/categóricas
# Tratamento dos valores ausentes pela moda
colunas_moda = [
    "education",
    "BPMeds"
]

for coluna in colunas_moda:
    moda = (
        df_silver
        .where(F.col(coluna).isNotNull())
        .groupBy(coluna)
        .count()
        .orderBy(F.desc("count"))
        .first()[0]
    )

    df_silver = df_silver.fillna({
        coluna: moda
    })

print("Tratamento de valores ausentes concluído.")

Tratamento de valores ausentes concluído.


#### Validação do tratamento

Após a imputação, é realizada uma nova verificação dos valores ausentes para confirmar a aplicação das regras definidas. Essa validação permite verificar se os atributos tratados deixaram de apresentar valores nulos antes da persistência dos dados na camada Silver.

In [0]:
# Verificação dos valores ausentes após o tratamento

expressoes_validacao = [
    F.sum(
        F.when(
            F.col(coluna).isNull(),
            1
        ).otherwise(0)
    ).alias(coluna)
    for coluna in df_silver.columns
]

df_validacao_ausentes = df_silver.select(
    expressoes_validacao
)

display(df_validacao_ausentes)

male,age,education,currentSmoker,cigsPerDay,BPMeds,prevalentStroke,prevalentHyp,diabetes,totChol,sysBP,diaBP,BMI,heartRate,glucose,TenYearCHD
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0


A validação após a imputação confirmou que não permaneceram valores ausentes no DataFrame destinado à camada Silver. Todas as 16 variáveis apresentaram contagem igual a zero para valores nulos.

Dessa forma, os valores ausentes identificados na camada Bronze foram tratados conforme as regras definidas: mediana para as variáveis numéricas contínuas e moda para `education` e `BPMeds`. O conjunto de dados está, portanto, preparado para as próximas etapas de transformação e validação da camada Silver.

### 4.2.4 Validação de valores e domínios

Após o tratamento dos valores ausentes e a padronização dos tipos, são realizadas verificações de consistência sobre os valores presentes nas variáveis.

O objetivo é identificar possíveis valores fora dos domínios esperados, como códigos inválidos em atributos binários ou valores numéricos incompatíveis com a natureza das variáveis, antes da persistência definitiva da camada Silver.

In [0]:
# Verificação dos valores distintos das variáveis categóricas/binárias

colunas_categoricas = [
    "male",
    "education",
    "currentSmoker",
    "BPMeds",
    "prevalentStroke",
    "prevalentHyp",
    "diabetes",
    "TenYearCHD"
]

for coluna in colunas_categoricas:
    print(f"\nValores distintos de {coluna}:")
    
    (
        df_silver
        .select(coluna)
        .distinct()
        .orderBy(coluna)
        .show()
    )


Valores distintos de male:
+----+
|male|
+----+
|   0|
|   1|
+----+


Valores distintos de education:
+---------+
|education|
+---------+
|      1.0|
|      2.0|
|      3.0|
|      4.0|
+---------+


Valores distintos de currentSmoker:
+-------------+
|currentSmoker|
+-------------+
|            0|
|            1|
+-------------+


Valores distintos de BPMeds:
+------+
|BPMeds|
+------+
|   0.0|
|   1.0|
+------+


Valores distintos de prevalentStroke:
+---------------+
|prevalentStroke|
+---------------+
|              0|
|              1|
+---------------+


Valores distintos de prevalentHyp:
+------------+
|prevalentHyp|
+------------+
|           0|
|           1|
+------------+


Valores distintos de diabetes:
+--------+
|diabetes|
+--------+
|       0|
|       1|
+--------+


Valores distintos de TenYearCHD:
+----------+
|TenYearCHD|
+----------+
|         0|
|         1|
+----------+



In [0]:
# Verificação de valores mínimos e máximos das variáveis numéricas

colunas_numericas = [
    "age",
    "cigsPerDay",
    "totChol",
    "sysBP",
    "diaBP",
    "BMI",
    "heartRate",
    "glucose"
]

expressoes_limites = []

for coluna in colunas_numericas:
    expressoes_limites.append(
        F.min(coluna).alias(f"{coluna}_min")
    )
    expressoes_limites.append(
        F.max(coluna).alias(f"{coluna}_max")
    )

display(
    df_silver.select(expressoes_limites)
)

age_min,age_max,cigsPerDay_min,cigsPerDay_max,totChol_min,totChol_max,sysBP_min,sysBP_max,diaBP_min,diaBP_max,BMI_min,BMI_max,heartRate_min,heartRate_max,glucose_min,glucose_max
32,70,0.0,70.0,107.0,696.0,83.5,295.0,48.0,142.5,15.54,56.8,44.0,143.0,40.0,394.0


A verificação dos domínios permitiu avaliar a consistência dos valores após as etapas de padronização e tratamento. As variáveis binárias apresentaram somente os valores 0 e 1, enquanto `education` apresentou as categorias 1, 2, 3 e 4, não sendo identificados códigos inesperados nos atributos analisados.

Nas variáveis numéricas, foram observadas amplitudes distintas, incluindo valores elevados em atributos como `totChol`, `sysBP` e `glucose`. Entretanto, esses valores não foram considerados automaticamente como erros, uma vez que podem representar observações extremas válidas do conjunto de dados.

Dessa forma, optou-se por preservar esses registros na camada Silver, evitando exclusões ou alterações sem evidência suficiente de inconsistência.

### 4.2.5 Validação final do DataFrame Silver

Após a aplicação das regras de tratamento e padronização, é realizada uma validação final do DataFrame destinado à camada Silver.

Nesta etapa são verificadas a quantidade de registros e atributos, a existência de registros duplicados e a presença de valores ausentes. O objetivo é confirmar a integridade do conjunto de dados antes de sua persistência.

In [0]:
# Validação final do DataFrame Silver

qtd_registros_silver = df_silver.count()
qtd_atributos_silver = len(df_silver.columns)

qtd_distintos_silver = df_silver.distinct().count()
qtd_duplicados_silver = qtd_registros_silver - qtd_distintos_silver

# Quantidade total de valores nulos
expressoes_nulos = [
    F.sum(
        F.when(F.col(coluna).isNull(), 1).otherwise(0)
    ).alias(coluna)
    for coluna in df_silver.columns
]

resultado_nulos = df_silver.select(expressoes_nulos).first()

total_nulos_silver = sum(resultado_nulos)

print(f"Quantidade de registros: {qtd_registros_silver}")
print(f"Quantidade de atributos: {qtd_atributos_silver}")
print(f"Quantidade de registros duplicados: {qtd_duplicados_silver}")
print(f"Quantidade total de valores ausentes: {total_nulos_silver}")

Quantidade de registros: 4238
Quantidade de atributos: 16
Quantidade de registros duplicados: 0
Quantidade total de valores ausentes: 0


A validação final confirmou a integridade do DataFrame após as etapas de tratamento e padronização. O conjunto permaneceu com 4.238 registros e 16 atributos, sem registros duplicados e sem valores ausentes.

Dessa forma, o DataFrame encontra-se preparado para ser persistido na camada Silver, preservando todos os registros e atributos da camada Bronze e incorporando os tratamentos de valores ausentes e a padronização dos tipos de dados realizados nesta etapa.

## 4.3 Persistência da camada Silver

Após a conclusão das etapas de tratamento, padronização e validação, o DataFrame resultante é persistido na camada Silver em formato Delta.

A tabela é armazenada no Unity Catalog, no catálogo `workspace` e schema `framingham_mvp`, permitindo que os dados tratados sejam utilizados nas etapas posteriores do pipeline sem alterar a tabela original da camada Bronze.

In [0]:
# Persistência dos dados na camada Silver

tabela_silver = "workspace.framingham_mvp.silver_framingham"

(
    df_silver.write
    .format("delta")
    .mode("overwrite")
    .saveAsTable(tabela_silver)
)

print(f"Tabela Silver criada: {tabela_silver}")

Tabela Silver criada: workspace.framingham_mvp.silver_framingham


A camada Silver foi persistida com sucesso no Unity Catalog, em formato Delta, na tabela `workspace.framingham_mvp.silver_framingham`.

Essa tabela contém a versão tratada e padronizada dos dados provenientes da camada Bronze, com os tipos de dados corrigidos e os valores ausentes tratados, preservando os 4.238 registros e 16 atributos do conjunto original. A camada Bronze permanece inalterada, garantindo a rastreabilidade dos dados ao longo do pipeline.

## 4.4 Validação da camada Silver persistida

Após a persistência, a tabela Silver é carregada novamente a partir do Unity Catalog para verificar se os dados foram armazenados corretamente.

Nesta validação são conferidas a quantidade de registros e atributos, a existência de registros duplicados e a presença de valores ausentes. Os resultados são comparados com aqueles obtidos antes da persistência, garantindo a integridade dos dados armazenados na camada Silver.

In [0]:
# Leitura da tabela Silver persistida
df_silver_validacao = spark.table(
    "workspace.framingham_mvp.silver_framingham"
)

# Quantidade de registros e atributos
registros_silver = df_silver_validacao.count()
atributos_silver = len(df_silver_validacao.columns)

# Verificação de duplicidades
registros_distintos_silver = df_silver_validacao.distinct().count()
duplicados_silver = registros_silver - registros_distintos_silver

# Verificação de valores ausentes
expressoes_nulos = [
    F.sum(
        F.when(F.col(coluna).isNull(), 1).otherwise(0)
    ).alias(coluna)
    for coluna in df_silver_validacao.columns
]

resultado_nulos = df_silver_validacao.select(
    expressoes_nulos
).first()

total_nulos = sum(resultado_nulos)

print(f"Registros na tabela Silver: {registros_silver}")
print(f"Atributos na tabela Silver: {atributos_silver}")
print(f"Registros duplicados: {duplicados_silver}")
print(f"Valores ausentes: {total_nulos}")

Registros na tabela Silver: 4238
Atributos na tabela Silver: 16
Registros duplicados: 0
Valores ausentes: 0


A validação da tabela Silver persistida confirmou a integridade dos dados após o armazenamento no Unity Catalog. A tabela apresentou 4.238 registros e 16 atributos, mantendo as mesmas dimensões verificadas no DataFrame antes da persistência.

Também não foram identificados registros duplicados ou valores ausentes. Dessa forma, confirma-se que a quantidade de registros e atributos e os resultados dos tratamentos de valores ausentes foram preservados corretamente na tabela `workspace.framingham_mvp.silver_framingham`.

##4.5  Visualização final da camada Silver

Após a validação da tabela persistida, são apresentados alguns registros da camada Silver para visualizar a estrutura final dos dados após as etapas de padronização e tratamento.

In [0]:
# Visualização da camada Silver persistida
display(df_silver_validacao.limit(10))

male,age,education,currentSmoker,cigsPerDay,BPMeds,prevalentStroke,prevalentHyp,diabetes,totChol,sysBP,diaBP,BMI,heartRate,glucose,TenYearCHD
1,39,4.0,0,0.0,0.0,0,0,0,195.0,106.0,70.0,26.97,80.0,77.0,0
0,46,2.0,0,0.0,0.0,0,0,0,250.0,121.0,81.0,28.73,95.0,76.0,0
1,48,1.0,1,20.0,0.0,0,0,0,245.0,127.5,80.0,25.34,75.0,70.0,0
0,61,3.0,1,30.0,0.0,0,1,0,225.0,150.0,95.0,28.58,65.0,103.0,1
0,46,3.0,1,23.0,0.0,0,0,0,285.0,130.0,84.0,23.1,85.0,85.0,0
0,43,2.0,0,0.0,0.0,0,1,0,228.0,180.0,110.0,30.3,77.0,99.0,0
0,63,1.0,0,0.0,0.0,0,0,0,205.0,138.0,71.0,33.11,60.0,85.0,1
0,45,2.0,1,20.0,0.0,0,0,0,313.0,100.0,71.0,21.68,79.0,78.0,0
1,52,1.0,0,0.0,0.0,0,1,0,260.0,141.5,89.0,26.36,76.0,79.0,0
1,43,1.0,1,30.0,0.0,0,1,0,225.0,162.0,107.0,23.61,93.0,88.0,0


A visualização apresenta uma amostra dos dados armazenados na camada Silver após os tratamentos realizados. O conjunto mantém os 16 atributos originais, agora com os tipos padronizados e os valores ausentes tratados conforme as regras definidas anteriormente.

# 5. Construção da camada Gold

A camada Gold representa a etapa de disponibilização dos dados já tratados e validados para consumo analítico e para as etapas posteriores de modelagem.

A partir da tabela Silver, serão realizadas análises relacionadas à estrutura e à distribuição das variáveis relevantes ao problema de negócio, com atenção especial à variável-alvo `TenYearCHD`, que representa a ocorrência de doença coronariana no horizonte de dez anos.

A construção desta camada busca disponibilizar um conjunto de dados confiável e adequado para análise e modelagem, mantendo a rastreabilidade em relação às camadas anteriores do pipeline.

## 5.1 Leitura da camada Silver

A tabela persistida na camada Silver é utilizada como fonte para a construção da camada Gold. Como os tratamentos de qualidade e padronização já foram realizados anteriormente, os dados são carregados a partir do Unity Catalog sem novas alterações neste primeiro momento.

In [0]:
# Leitura da camada Silver

df_gold_base = spark.table(
    "workspace.framingham_mvp.silver_framingham"
)

print(f"Quantidade de registros: {df_gold_base.count()}")
print(f"Quantidade de atributos: {len(df_gold_base.columns)}")

Quantidade de registros: 4238
Quantidade de atributos: 16


A leitura da camada Silver disponibilizou 4.238 registros e 16 atributos para a construção da camada Gold, correspondentes à estrutura previamente validada após sua persistência no Unity Catalog.

A partir deste ponto, serão realizadas análises voltadas à preparação dos dados para consumo analítico e modelagem, com atenção especial à variável-alvo `TenYearCHD`.

## 5.2 Análise da variável-alvo

A variável `TenYearCHD` representa a ocorrência de doença coronariana no horizonte de dez anos e constitui a variável-alvo considerada nas etapas posteriores de modelagem.

Antes da preparação dos dados para os modelos, é analisada a distribuição das classes dessa variável. Essa verificação permite identificar a quantidade e a proporção de registros associados a cada classe e avaliar a existência de desbalanceamento no conjunto de dados.

Nesta etapa, não são realizadas alterações nos registros, sendo efetuada apenas a análise da distribuição da variável-alvo.

In [0]:
from pyspark.sql import functions as F

# Quantidade total de registros
total_registros = df_gold_base.count()

# Distribuição da variável-alvo
distribuicao_target = (
    df_gold_base
    .groupBy("TenYearCHD")
    .count()
    .withColumn(
        "percentual",
        F.round(
            (F.col("count") / total_registros) * 100,
            2
        )
    )
    .orderBy("TenYearCHD")
)

display(distribuicao_target)

TenYearCHD,count,percentual
0,3594,84.8
1,644,15.2


A análise da variável-alvo `TenYearCHD` identificou uma distribuição desigual entre as duas classes. Dos 4.238 registros do conjunto de dados, 3.594 (84,8%) pertencem à classe 0, enquanto 644 (15,2%) pertencem à classe 1.

A classe 0 representa os indivíduos sem ocorrência de doença coronariana no horizonte de dez anos, enquanto a classe 1 representa os indivíduos com ocorrência nesse período.

Os resultados evidenciam um desbalanceamento entre as classes da variável-alvo, com menor representação da classe 1. Essa característica deverá ser considerada nas etapas posteriores de modelagem e avaliação.

Nesta etapa, nenhuma técnica de balanceamento é aplicada, preservando-se a distribuição observada nos dados para as análises e preparações subsequentes.

## 5.3 Análise das variáveis numéricas

Após a análise da variável-alvo, são avaliadas as principais variáveis explicativas do conjunto de dados. O objetivo desta etapa é compreender a distribuição e a amplitude dos atributos que poderão ser utilizados posteriormente na modelagem.

Para as variáveis numéricas contínuas, são calculadas estatísticas descritivas como média, desvio-padrão, valores mínimo e máximo e quartis. Essa análise permite observar diferenças de escala, características das distribuições e possíveis valores extremos.

Nesta etapa, nenhuma transformação é realizada, sendo efetuada apenas a análise das características das variáveis disponíveis na camada Silver.

In [0]:
from pyspark.sql import functions as F

# Variáveis numéricas contínuas
colunas_numericas = [
    "age",
    "cigsPerDay",
    "totChol",
    "sysBP",
    "diaBP",
    "BMI",
    "heartRate",
    "glucose"
]

# Estatísticas descritivas
estatisticas_numericas = (
    df_gold_base
    .select(colunas_numericas)
    .summary(
        "count",
        "mean",
        "stddev",
        "min",
        "25%",
        "50%",
        "75%",
        "max"
    )
)

display(estatisticas_numericas)

summary,age,cigsPerDay,totChol,sysBP,diaBP,BMI,heartRate,glucose
count,4238,4238,4238,4238,4238,4238,4238,4238
mean,49.58494572911751,8.941481831052384,236.68947616800378,132.35240679565834,82.89346389806512,25.80011562057573,75.87871637564889,81.60358659745162
stddev,8.572159925118484,11.90239876710421,44.327426965294144,22.038096643544584,11.910849600225502,4.071050198070549,12.02518461212337,22.86524554289789
min,32,0.0,107.0,83.5,48.0,15.54,44.0,40.0
25%,42,0.0,206.0,117.0,75.0,23.08,68.0,72.0
50%,49,0.0,234.0,128.0,82.0,25.38,75.0,78.0
75%,56,20.0,262.0,144.0,90.0,28.04,83.0,85.0
max,70,70.0,696.0,295.0,142.5,56.8,143.0,394.0


A análise das estatísticas descritivas confirmou a presença de 4.238 observações em todas as variáveis numéricas avaliadas.

As variáveis apresentam diferentes amplitudes e características de distribuição. Em particular, `totChol`, `sysBP` e `glucose` apresentam valores máximos consideravelmente superiores aos respectivos quartis superiores e medianas. Também são observadas amplitudes relevantes em atributos como `BMI` e `heartRate`.

Esses resultados indicam a presença de observações de maior magnitude em algumas variáveis. Entretanto, tais valores não são considerados automaticamente inconsistências, pois podem representar condições válidas da população analisada. Por esse motivo, os registros são preservados nesta etapa, evitando exclusões sem evidência suficiente de erro.

As características de escala e distribuição observadas poderão ser consideradas posteriormente, de acordo com as necessidades das técnicas de modelagem utilizadas.

## 5.4 Análise das variáveis categóricas e binárias

Além das variáveis numéricas contínuas, o conjunto de dados contém atributos categóricos e binários representados numericamente. Nesta etapa, é analisada a distribuição das categorias presentes nesses atributos, com o objetivo de compreender sua composição antes da preparação dos dados para modelagem.

São avaliadas as variáveis `male`, `education`, `currentSmoker`, `BPMeds`, `prevalentStroke`, `prevalentHyp` e `diabetes`. A análise considera a quantidade de registros associada a cada categoria, permitindo identificar categorias pouco frequentes e diferenças relevantes em suas distribuições.

Nesta etapa, os dados são apenas analisados, sem aplicação de novas transformações.

In [0]:
from pyspark.sql import functions as F

# Variáveis categóricas e binárias
colunas_categoricas = [
    "male",
    "education",
    "currentSmoker",
    "BPMeds",
    "prevalentStroke",
    "prevalentHyp",
    "diabetes"
]

# Distribuição de cada variável
for coluna in colunas_categoricas:
    print(f"Distribuição de {coluna}:")

    distribuicao = (
        df_gold_base
        .groupBy(coluna)
        .count()
        .withColumn(
            "percentual",
            F.round(
                (F.col("count") / df_gold_base.count()) * 100,
                2
            )
        )
        .orderBy(coluna)
    )

    display(distribuicao)

Distribuição de male:


male,count,percentual
0,2419,57.08
1,1819,42.92


Distribuição de education:


education,count,percentual
1.0,1825,43.06
2.0,1253,29.57
3.0,687,16.21
4.0,473,11.16


Distribuição de currentSmoker:


currentSmoker,count,percentual
0,2144,50.59
1,2094,49.41


Distribuição de BPMeds:


BPMeds,count,percentual
0.0,4114,97.07
1.0,124,2.93


Distribuição de prevalentStroke:


prevalentStroke,count,percentual
0,4213,99.41
1,25,0.59


Distribuição de prevalentHyp:


prevalentHyp,count,percentual
0,2922,68.95
1,1316,31.05


Distribuição de diabetes:


diabetes,count,percentual
0,4129,97.43
1,109,2.57


A análise das variáveis categóricas e binárias permitiu observar diferenças na frequência das categorias presentes no conjunto de dados.

Na variável `male`, 57,08% dos registros pertencem à categoria 0 e 42,92% à categoria 1. A variável `currentSmoker` apresenta distribuição próxima entre as categorias, com 50,59% dos registros na categoria 0 e 49,41% na categoria 1.

Para `education`, observa-se maior concentração nas categorias 1 e 2, que representam, respectivamente, 43,06% e 29,57% dos registros. As categorias 3 e 4 correspondem a 16,21% e 11,16% do conjunto.

Entre as demais variáveis binárias, `prevalentHyp` apresenta 31,05% dos registros na categoria 1. Já `BPMeds`, `prevalentStroke` e `diabetes` apresentam menor frequência da categoria 1, correspondendo a 2,93%, 0,59% e 2,57% dos registros, respectivamente.

Os resultados mostram que algumas características possuem distribuição relativamente equilibrada entre suas categorias, enquanto outras apresentam baixa frequência da categoria 1. Essas diferenças deverão ser consideradas nas etapas posteriores de análise e modelagem.

Nesta etapa, nenhuma categoria é removida ou modificada, preservando-se as informações provenientes da camada Silver.

## 5.5 Relação das variáveis com a variável-alvo

Após a análise individual das variáveis, é avaliada sua relação com a variável-alvo `TenYearCHD`. O objetivo é identificar diferenças descritivas entre os indivíduos das classes 0 e 1, contribuindo para a compreensão das características associadas à ocorrência de doença coronariana no horizonte de dez anos.

Inicialmente, são comparadas as médias das principais variáveis numéricas entre as duas classes da variável-alvo. Essa análise possui caráter exploratório e não estabelece relações de causalidade, sendo utilizada para identificar padrões que poderão ser considerados nas etapas posteriores de modelagem.

In [0]:
from pyspark.sql import functions as F

# Variáveis numéricas para comparação entre as classes
colunas_numericas = [
    "age",
    "cigsPerDay",
    "totChol",
    "sysBP",
    "diaBP",
    "BMI",
    "heartRate",
    "glucose"
]

# Média das variáveis numéricas por classe da variável-alvo
expressoes_media = [
    F.round(F.avg(coluna), 2).alias(f"media_{coluna}")
    for coluna in colunas_numericas
]

comparacao_target = (
    df_gold_base
    .groupBy("TenYearCHD")
    .agg(*expressoes_media)
    .orderBy("TenYearCHD")
)

display(comparacao_target)

TenYearCHD,media_age,media_cigsPerDay,media_totChol,media_sysBP,media_diaBP,media_BMI,media_heartRate,media_glucose
0,48.77,8.64,235.16,130.33,82.16,25.67,75.76,80.43
1,54.15,10.6,245.23,143.62,86.98,26.51,76.53,88.15


A comparação das médias das variáveis numéricas entre as classes de `TenYearCHD` revelou diferenças descritivas entre os grupos.

Os indivíduos pertencentes à classe 1 apresentaram, em média, maior idade (54,15 anos) em comparação à classe 0 (48,77 anos). Também foram observadas médias superiores de consumo diário de cigarros (10,60 contra 8,64), colesterol total (245,23 contra 235,16), pressão arterial sistólica (143,62 contra 130,33) e pressão arterial diastólica (86,98 contra 82,16).

Diferenças também foram observadas em `BMI`, com médias de 26,51 e 25,67, em `heartRate`, com 76,53 e 75,76, e em `glucose`, com 88,15 e 80,43, respectivamente para as classes 1 e 0.

De forma geral, todas as variáveis numéricas analisadas apresentaram médias superiores entre os registros pertencentes à classe 1 de `TenYearCHD`. As diferenças mais evidentes foram observadas em idade, pressão arterial sistólica e glicose. Esses resultados possuem caráter descritivo e não devem ser interpretados isoladamente como evidência de relação causal.

## 5.6 Organização do conjunto de dados para análise e modelagem

Com base nas análises realizadas, é organizado o conjunto de dados destinado à camada Gold. Nesta etapa, são definidas explicitamente as variáveis explicativas e a variável-alvo que serão disponibilizadas para as etapas posteriores de análise e modelagem.

A variável `TenYearCHD` é mantida como variável-alvo, enquanto os demais atributos são preservados como potenciais variáveis explicativas. Não são realizadas exclusões baseadas apenas nas análises descritivas anteriores, evitando a remoção prematura de informações potencialmente relevantes.

A organização mantém os 4.238 registros provenientes da camada Silver e estrutura o conjunto para sua disponibilização na camada Gold.

In [0]:
# Variável-alvo
coluna_target = "TenYearCHD"

# Variáveis explicativas
colunas_features = [
    coluna
    for coluna in df_gold_base.columns
    if coluna != coluna_target
]

# Construção do DataFrame destinado à camada Gold
df_gold = df_gold_base.select(
    *colunas_features,
    coluna_target
)

print(f"Quantidade de registros: {df_gold.count()}")
print(f"Quantidade de variáveis explicativas: {len(colunas_features)}")
print(f"Variável-alvo: {coluna_target}")

display(df_gold.limit(10))

Quantidade de registros: 4238
Quantidade de variáveis explicativas: 15
Variável-alvo: TenYearCHD


male,age,education,currentSmoker,cigsPerDay,BPMeds,prevalentStroke,prevalentHyp,diabetes,totChol,sysBP,diaBP,BMI,heartRate,glucose,TenYearCHD
1,39,4.0,0,0.0,0.0,0,0,0,195.0,106.0,70.0,26.97,80.0,77.0,0
0,46,2.0,0,0.0,0.0,0,0,0,250.0,121.0,81.0,28.73,95.0,76.0,0
1,48,1.0,1,20.0,0.0,0,0,0,245.0,127.5,80.0,25.34,75.0,70.0,0
0,61,3.0,1,30.0,0.0,0,1,0,225.0,150.0,95.0,28.58,65.0,103.0,1
0,46,3.0,1,23.0,0.0,0,0,0,285.0,130.0,84.0,23.1,85.0,85.0,0
0,43,2.0,0,0.0,0.0,0,1,0,228.0,180.0,110.0,30.3,77.0,99.0,0
0,63,1.0,0,0.0,0.0,0,0,0,205.0,138.0,71.0,33.11,60.0,85.0,1
0,45,2.0,1,20.0,0.0,0,0,0,313.0,100.0,71.0,21.68,79.0,78.0,0
1,52,1.0,0,0.0,0.0,0,1,0,260.0,141.5,89.0,26.36,76.0,79.0,0
1,43,1.0,1,30.0,0.0,0,1,0,225.0,162.0,107.0,23.61,93.0,88.0,0


A organização do conjunto destinado à camada Gold manteve os 4.238 registros disponíveis na camada Silver. Dos 16 atributos presentes no conjunto, 15 foram definidos como potenciais variáveis explicativas e `TenYearCHD` foi mantida como variável-alvo.

Nesta etapa, não foram removidos registros nem atributos, preservando as informações previamente tratadas e validadas. O conjunto resultante encontra-se estruturado para sua persistência na camada Gold e posterior utilização nas etapas de análise e modelagem.

## 5.7 Persistência da camada Gold

Após a organização das variáveis explicativas e da variável-alvo, o conjunto de dados resultante é persistido na camada Gold.

A persistência disponibiliza uma versão consolidada e rastreável dos dados para consumo analítico e para utilização nas etapas posteriores de modelagem, mantendo a separação entre as diferentes camadas da arquitetura de dados.

A tabela Gold é armazenada no Unity Catalog em formato Delta, utilizando como origem os dados previamente tratados e validados na camada Silver.

In [0]:
# Nome da tabela da camada Gold
tabela_gold = "workspace.framingham_mvp.gold_framingham"

# Persistência da camada Gold no Unity Catalog
(
    df_gold.write
    .format("delta")
    .mode("overwrite")
    .saveAsTable(tabela_gold)
)

print(f"Tabela Gold persistida com sucesso: {tabela_gold}")

Tabela Gold persistida com sucesso: workspace.framingham_mvp.gold_framingham


A persistência da camada Gold foi realizada com sucesso no Unity Catalog, por meio da tabela `workspace.framingham_mvp.gold_framingham`, utilizando o formato Delta.

Com essa etapa, o conjunto de dados organizado a partir da camada Silver passa a estar disponível de forma persistente para consumo analítico e para utilização nas etapas posteriores de modelagem, mantendo a rastreabilidade do fluxo de dados entre as camadas Silver e Gold.

Na etapa seguinte, a tabela Gold será carregada novamente a partir do Unity Catalog para validação de sua integridade após a persistência.

## 5.8 Validação da camada Gold persistida

Após a persistência, a tabela Gold é carregada novamente a partir do Unity Catalog para verificar se os dados foram armazenados corretamente.

Nesta validação são conferidas a quantidade de registros e atributos, a existência de registros duplicados e a presença de valores ausentes. Os resultados são comparados com a estrutura preparada anteriormente, permitindo verificar a integridade do conjunto disponibilizado na camada Gold.

In [0]:
from pyspark.sql import functions as F

# Leitura da tabela Gold persistida
df_gold_validacao = spark.table(
    "workspace.framingham_mvp.gold_framingham"
)

# Quantidade de registros e atributos
registros_gold = df_gold_validacao.count()
atributos_gold = len(df_gold_validacao.columns)

# Verificação de duplicidades
registros_distintos_gold = df_gold_validacao.distinct().count()
duplicados_gold = registros_gold - registros_distintos_gold

# Verificação de valores ausentes
expressoes_nulos = [
    F.sum(
        F.when(F.col(coluna).isNull(), 1).otherwise(0)
    ).alias(coluna)
    for coluna in df_gold_validacao.columns
]

resultado_nulos = (
    df_gold_validacao
    .select(expressoes_nulos)
    .first()
)

total_nulos = sum(resultado_nulos)

print(f"Registros na tabela Gold: {registros_gold}")
print(f"Atributos na tabela Gold: {atributos_gold}")
print(f"Registros duplicados: {duplicados_gold}")
print(f"Valores ausentes: {total_nulos}")

Registros na tabela Gold: 4238
Atributos na tabela Gold: 16
Registros duplicados: 0
Valores ausentes: 0


A validação da camada Gold persistida confirmou a manutenção dos 4.238 registros e dos 16 atributos presentes no conjunto organizado anteriormente.

Não foram identificados registros duplicados nem valores ausentes após a persistência, confirmando que a gravação no Unity Catalog preservou a estrutura e a integridade do conjunto de dados.

Dessa forma, a camada Gold encontra-se consolidada e disponível para consumo analítico e para utilização nas etapas posteriores de modelagem.

##5.9 Visualização final da camada Gold

Após a validação da persistência, são apresentados alguns registros da camada Gold para visualizar a estrutura do conjunto disponibilizado para consumo analítico e para as etapas posteriores de modelagem.

In [0]:
# Visualização da camada Gold persistida
display(df_gold_validacao.limit(10))

male,age,education,currentSmoker,cigsPerDay,BPMeds,prevalentStroke,prevalentHyp,diabetes,totChol,sysBP,diaBP,BMI,heartRate,glucose,TenYearCHD
1,39,4.0,0,0.0,0.0,0,0,0,195.0,106.0,70.0,26.97,80.0,77.0,0
0,46,2.0,0,0.0,0.0,0,0,0,250.0,121.0,81.0,28.73,95.0,76.0,0
1,48,1.0,1,20.0,0.0,0,0,0,245.0,127.5,80.0,25.34,75.0,70.0,0
0,61,3.0,1,30.0,0.0,0,1,0,225.0,150.0,95.0,28.58,65.0,103.0,1
0,46,3.0,1,23.0,0.0,0,0,0,285.0,130.0,84.0,23.1,85.0,85.0,0
0,43,2.0,0,0.0,0.0,0,1,0,228.0,180.0,110.0,30.3,77.0,99.0,0
0,63,1.0,0,0.0,0.0,0,0,0,205.0,138.0,71.0,33.11,60.0,85.0,1
0,45,2.0,1,20.0,0.0,0,0,0,313.0,100.0,71.0,21.68,79.0,78.0,0
1,52,1.0,0,0.0,0.0,0,1,0,260.0,141.5,89.0,26.36,76.0,79.0,0
1,43,1.0,1,30.0,0.0,0,1,0,225.0,162.0,107.0,23.61,93.0,88.0,0


A visualização apresenta uma amostra do conjunto final disponibilizado na camada Gold. A tabela contém as 15 variáveis explicativas e a variável-alvo `TenYearCHD`, totalizando 16 atributos organizados para utilização nas etapas posteriores de análise e modelagem.

# 6. Análise da qualidade dos dados

A qualidade dos dados foi avaliada ao longo das diferentes etapas de construção do pipeline. Durante o tratamento e a validação das camadas Silver e Gold, foram realizadas verificações relacionadas à presença de valores ausentes, registros duplicados, consistência dos valores e preservação da estrutura do conjunto de dados.

Nesta etapa, os resultados dessas verificações são consolidados e complementados com uma análise específica das observações de maior magnitude, buscando avaliar a qualidade do conjunto final disponibilizado na camada Gold e sua adequação para as análises e etapas posteriores de modelagem.

## 6.1 Consolidação das verificações de qualidade

As verificações realizadas nas etapas anteriores demonstraram que os tratamentos aplicados durante a construção da camada Silver resultaram em um conjunto sem valores ausentes e sem registros duplicados.

Após a persistência das camadas Silver e Gold, os dados foram carregados novamente a partir do Unity Catalog e submetidos a novas validações. Na validação final da camada Gold foram encontrados 4.238 registros e 16 atributos, sem ocorrência de registros duplicados ou valores ausentes.

Dessa forma, não é necessário realizar novos tratamentos relacionados a valores ausentes ou duplicidades nesta etapa. As verificações seguintes complementam as análises já realizadas ao longo do pipeline, com atenção aos valores presentes nas variáveis e às observações de maior magnitude.

## 6.2 Consistência dos valores e domínios

A consistência dos valores foi verificada durante a construção da camada Silver. As variáveis binárias apresentaram somente as categorias esperadas, 0 e 1, enquanto `education` apresentou as categorias 1, 2, 3 e 4. Não foram identificados códigos inesperados nessas variáveis.

Também foram analisados os intervalos das variáveis numéricas. Foram observados valores entre 32 e 70 anos para `age`, 0 e 70 para `cigsPerDay`, 107 e 696 para `totChol`, 83,5 e 295 para `sysBP`, 48 e 142,5 para `diaBP`, 15,54 e 56,8 para `BMI`, 44 e 143 para `heartRate` e 40 e 394 para `glucose`.

Embora alguns atributos apresentem valores máximos elevados, principalmente `totChol`, `sysBP`, `BMI` e `glucose`, esses registros não foram classificados automaticamente como inconsistentes, pois a magnitude isolada não é suficiente para caracterizar um erro.

Como essas verificações já foram realizadas sobre os dados tratados e a persistência posterior confirmou a preservação do conjunto nas camadas Silver e Gold, não foram necessárias novas correções relacionadas aos domínios das variáveis.


In [0]:
from pyspark.sql import functions as F

# Variáveis binárias
colunas_binarias = [
    "male",
    "currentSmoker",
    "BPMeds",
    "prevalentStroke",
    "prevalentHyp",
    "diabetes",
    "TenYearCHD"
]

print("Valores encontrados nas variáveis binárias:")

for coluna in colunas_binarias:
    valores = [
        linha[coluna]
        for linha in (
            df_gold_validacao
            .select(coluna)
            .distinct()
            .orderBy(coluna)
            .collect()
        )
    ]
    print(f"{coluna}: {valores}")

# Categorias de education
valores_education = [
    linha["education"]
    for linha in (
        df_gold_validacao
        .select("education")
        .distinct()
        .orderBy("education")
        .collect()
    )
]

print(f"\neducation: {valores_education}")

# Variáveis numéricas contínuas
colunas_numericas = [
    "age",
    "cigsPerDay",
    "totChol",
    "sysBP",
    "diaBP",
    "BMI",
    "heartRate",
    "glucose"
]

# Valores mínimos e máximos
faixas_numericas = df_gold_validacao.select(
    *[
        expressao
        for coluna in colunas_numericas
        for expressao in [
            F.min(coluna).alias(f"{coluna}_min"),
            F.max(coluna).alias(f"{coluna}_max")
        ]
    ]
)

display(faixas_numericas)

Valores encontrados nas variáveis binárias:
male: [0, 1]
currentSmoker: [0, 1]
BPMeds: [0.0, 1.0]
prevalentStroke: [0, 1]
prevalentHyp: [0, 1]
diabetes: [0, 1]
TenYearCHD: [0, 1]

education: [1.0, 2.0, 3.0, 4.0]


age_min,age_max,cigsPerDay_min,cigsPerDay_max,totChol_min,totChol_max,sysBP_min,sysBP_max,diaBP_min,diaBP_max,BMI_min,BMI_max,heartRate_min,heartRate_max,glucose_min,glucose_max
32,70,0,70,107,696,83.5,295,48,142.5,15.54,56.8,44,143,40,394


A verificação das variáveis categóricas e binárias mostrou que os valores encontrados estão de acordo com a estrutura esperada do conjunto de dados. As variáveis binárias apresentam exclusivamente os valores 0 e 1, enquanto a variável `education` apresenta as categorias 1, 2, 3 e 4.

Nas variáveis numéricas, foram observadas diferentes amplitudes entre os valores mínimos e máximos. Algumas variáveis apresentam valores máximos consideravelmente distantes dos valores centrais observados anteriormente, como `totChol`, `sysBP`, `BMI` e `glucose`.

Esses valores não são considerados erros apenas por apresentarem maior magnitude. Dessa forma, será realizada uma análise específica de valores extremos para identificar observações que se encontram distantes da distribuição predominante dos dados.

## 6.3 Verificação de valores extremos

A partir dos intervalos observados anteriormente, algumas variáveis numéricas apresentaram valores máximos consideravelmente superiores aos valores centrais do conjunto de dados. Para complementar essa verificação, são apresentados os registros com os maiores valores das variáveis `totChol`, `sysBP`, `BMI` e `glucose`.

O objetivo desta análise é verificar diretamente as observações mais extremas do conjunto de dados, sem classificá-las automaticamente como erros ou realizar sua remoção.

In [0]:
from pyspark.sql import functions as F

# Variáveis que apresentaram valores máximos elevados
variaveis_extremos = [
    "totChol",
    "sysBP",
    "BMI",
    "glucose"
]

# Maiores valores observados em cada variável
for coluna in variaveis_extremos:
    print(f"Maiores valores de {coluna}:")

    display(
        df_gold_validacao
        .select(coluna, "TenYearCHD")
        .orderBy(F.col(coluna).desc())
        .limit(5)
    )

Maiores valores de totChol:


totChol,TenYearCHD
696.0,0
600.0,1
464.0,1
453.0,0
439.0,1


Maiores valores de sysBP:


sysBP,TenYearCHD
295.0,1
248.0,1
244.0,1
243.0,0
235.0,1


Maiores valores de BMI:


BMI,TenYearCHD
56.8,1
51.28,0
45.8,0
45.79,0
44.71,0


Maiores valores de glucose:


glucose,TenYearCHD
394.0,1
394.0,1
386.0,0
370.0,0
368.0,1


A verificação dos maiores valores confirmou a presença de observações com magnitudes elevadas nas variáveis analisadas. Foram encontrados valores de até 696 para `totChol`, 295 para `sysBP`, 56,8 para `BMI` e 394 para `glucose`.

A comparação com a variável-alvo também mostra que valores elevados aparecem tanto em registros com `TenYearCHD = 0` quanto com `TenYearCHD = 1`. No caso de `sysBP`, quatro dos cinco maiores valores estão associados à classe 1, enquanto nas demais variáveis a distribuição entre as classes apresenta comportamentos distintos. Essa observação possui caráter apenas descritivo e não permite estabelecer uma relação causal.

Apesar de elevados, esses valores não foram considerados erros apenas com base em sua magnitude, pois podem representar condições clínicas reais. Como não foram identificadas evidências suficientes para classificá-los como registros inválidos, nenhuma observação foi removida ou alterada nesta etapa.

Dessa forma, os valores foram preservados no conjunto de dados. Entretanto, sua presença deve ser considerada nas etapas posteriores de análise e modelagem, uma vez que observações extremas podem influenciar determinadas técnicas estatísticas e algoritmos.

## 6.4 Síntese da qualidade dos dados

Após as verificações realizadas, é possível consolidar os resultados da análise de qualidade do conjunto de dados disponibilizado na camada Gold.

Foram avaliadas a presença de valores ausentes e registros duplicados, a adequação dos valores encontrados nas variáveis categóricas e binárias, os intervalos das variáveis numéricas e a ocorrência de valores extremos. Essas verificações permitem avaliar se existem problemas que possam comprometer as análises e as etapas posteriores de modelagem.

A síntese a seguir apresenta os principais resultados obtidos durante essa avaliação.

In [0]:
# Síntese das verificações de qualidade dos dados

resumo_qualidade = [
    ("Valores ausentes", "Não identificados", "0 valores ausentes"),
    ("Registros duplicados", "Não identificados", "0 registros duplicados"),
    ("Variáveis binárias", "Valores adequados", "Valores 0 e 1"),
    ("Variável education", "Valores adequados", "Categorias 1, 2, 3 e 4"),
    ("Variáveis numéricas", "Intervalos verificados", "Valores mínimos e máximos analisados"),
    ("Valores extremos", "Mantidos", "Sem evidência suficiente para classificá-los como erros")
]

df_resumo_qualidade = spark.createDataFrame(
    resumo_qualidade,
    ["verificacao", "resultado", "observacao"]
)

display(df_resumo_qualidade)

verificacao,resultado,observacao
Valores ausentes,Não identificados,0 valores ausentes
Registros duplicados,Não identificados,0 registros duplicados
Variáveis binárias,Valores adequados,Valores 0 e 1
Variável education,Valores adequados,"Categorias 1, 2, 3 e 4"
Variáveis numéricas,Intervalos verificados,Valores mínimos e máximos analisados
Valores extremos,Mantidos,Sem evidência suficiente para classificá-los como erros


A síntese das verificações indica que o conjunto de dados disponibilizado na camada Gold não apresenta valores ausentes ou registros duplicados e mantém os valores categóricos e binários dentro dos domínios esperados.

Os intervalos das variáveis numéricas também foram verificados e os valores de maior magnitude foram preservados, pois não foram encontradas evidências suficientes para caracterizá-los como erros. Esses valores, entretanto, devem ser considerados durante a interpretação das análises e nas etapas posteriores de modelagem.

Dessa forma, o conjunto final mantém 4.238 registros e 16 atributos, encontrando-se estruturado e validado para continuidade das análises.

# 7. Análise final dos dados e respostas às perguntas de negócio

Após a construção, tratamento e validação do pipeline, os dados disponibilizados na camada Gold são utilizados para responder às cinco perguntas de negócio definidas na seção 1.3.

Parte das informações necessárias já foi obtida durante as análises realizadas nas etapas anteriores. Dessa forma, esses resultados são retomados quando aplicáveis, evitando a repetição desnecessária de cálculos. Novas análises são realizadas apenas quando necessárias para responder diretamente às questões propostas.

Os resultados apresentados possuem caráter descritivo e representam associações observadas no conjunto de dados, não permitindo estabelecer relações de causalidade.

## 7.1 Ocorrência de doença coronariana por faixa etária

A primeira pergunta de negócio busca verificar como a ocorrência de doença coronariana em dez anos varia de acordo com a faixa etária dos pacientes.

Embora a seção 5.5 tenha mostrado que a idade média é maior entre os indivíduos com `TenYearCHD = 1`, nesta análise os pacientes são agrupados em faixas etárias para permitir uma comparação mais direta da frequência da classe positiva entre diferentes grupos de idade.

In [0]:
from pyspark.sql import functions as F

# Criação das faixas etárias
df_faixa_etaria = (
    df_gold_validacao
    .withColumn(
        "faixa_etaria",
        F.when(F.col("age") < 40, "30-39")
         .when(F.col("age") < 50, "40-49")
         .when(F.col("age") < 60, "50-59")
         .otherwise("60-70")
    )
)

# Ocorrência de TenYearCHD por faixa etária
resultado_idade = (
    df_faixa_etaria
    .groupBy("faixa_etaria")
    .agg(
        F.count("*").alias("total_registros"),
        F.sum("TenYearCHD").alias("casos_CHD")
    )
    .withColumn(
        "percentual_CHD",
        F.round(
            (F.col("casos_CHD") / F.col("total_registros")) * 100,
            2
        )
    )
    .orderBy("faixa_etaria")
)

display(resultado_idade)

faixa_etaria,total_registros,casos_CHD,percentual_CHD
30-39,555,23,4.14
40-49,1660,167,10.06
50-59,1333,263,19.73
60-70,690,191,27.68


Os resultados mostram um aumento progressivo na ocorrência de `TenYearCHD = 1` conforme a faixa etária dos pacientes aumenta. Entre os indivíduos de 30 a 39 anos, 4,14% apresentaram ocorrência de doença coronariana no horizonte de dez anos. Esse percentual aumenta para 10,06% entre 40 e 49 anos, 19,73% entre 50 e 59 anos e atinge 27,68% entre 60 e 70 anos.

Dessa forma, no conjunto de dados analisado, observa-se uma associação entre faixas etárias mais elevadas e maior frequência de ocorrência de doença coronariana em dez anos. Esse resultado é consistente com a análise apresentada na seção 5.5, na qual os indivíduos com `TenYearCHD = 1` também apresentaram maior idade média.

## 7.2 Ocorrência de doença coronariana e hipertensão

A segunda pergunta de negócio busca verificar se pacientes com hipertensão apresentam maior ocorrência de doença coronariana em dez anos.

Para responder à questão, é comparada a proporção de registros com `TenYearCHD = 1` entre indivíduos sem hipertensão (`prevalentHyp = 0`) e com hipertensão (`prevalentHyp = 1`).

In [0]:
resultado_hipertensao = (
    df_gold_validacao
    .groupBy("prevalentHyp")
    .agg(
        F.count("*").alias("total_registros"),
        F.sum("TenYearCHD").alias("casos_CHD")
    )
    .withColumn(
        "percentual_CHD",
        F.round(
            (F.col("casos_CHD") / F.col("total_registros")) * 100,
            2
        )
    )
    .orderBy("prevalentHyp")
)

display(resultado_hipertensao)

prevalentHyp,total_registros,casos_CHD,percentual_CHD
0,2922,319,10.92
1,1316,325,24.7


Os resultados mostram que, entre os indivíduos sem hipertensão, 10,92% apresentaram `TenYearCHD = 1`. Entre os indivíduos com hipertensão, esse percentual foi de 24,70%.

Dessa forma, no conjunto de dados analisado, observa-se uma associação entre a presença de hipertensão e maior frequência de ocorrência de doença coronariana no horizonte de dez anos. A proporção observada no grupo com hipertensão foi superior à encontrada entre os indivíduos sem hipertensão.

## 7.3 Indicadores clínicos e ocorrência de doença coronariana

A terceira pergunta de negócio busca identificar como indicadores clínicos como colesterol, pressão arterial, IMC e glicose diferem entre pacientes com e sem ocorrência de doença coronariana em dez anos.

Essa comparação já foi realizada na seção 5.5, na qual foram calculadas as médias das variáveis numéricas para as classes de `TenYearCHD`. Dessa forma, os resultados obtidos anteriormente são reutilizados para responder à questão, evitando a repetição do mesmo processamento.

Para facilitar a comparação entre os grupos, os resultados dos indicadores clínicos são organizados a seguir em um formato apropriado para visualização.

In [0]:
# Seleção dos indicadores clínicos analisados na pergunta de negócio
indicadores_clinicos = comparacao_target.select(
    "TenYearCHD",
    "media_totChol",
    "media_sysBP",
    "media_diaBP",
    "media_BMI",
    "media_glucose"
)

display(indicadores_clinicos)

TenYearCHD,media_totChol,media_sysBP,media_diaBP,media_BMI,media_glucose
0,235.16,130.33,82.16,25.67,80.43
1,245.23,143.62,86.98,26.51,88.15


Os pacientes com `TenYearCHD = 1` apresentaram colesterol total médio de 245,23, enquanto na classe 0 a média foi de 235,16. A pressão arterial sistólica apresentou médias de 143,62 e 130,33, respectivamente, enquanto a pressão arterial diastólica apresentou médias de 86,98 e 82,16.

O IMC médio foi de 26,51 entre os pacientes com `TenYearCHD = 1` e de 25,67 entre aqueles da classe 0. Para a glicose, foram observadas médias de 88,15 na classe 1 e 80,43 na classe 0.

Assim, todos os indicadores clínicos considerados apresentaram médias mais elevadas entre os pacientes com ocorrência de doença coronariana em dez anos. A maior diferença absoluta entre as médias foi observada na pressão arterial sistólica, seguida pelo colesterol total e pela glicose, enquanto o IMC apresentou a menor diferença entre as classes.

## 7.4 Tabagismo e ocorrência de doença coronariana

A quarta pergunta de negócio busca analisar como fatores relacionados ao tabagismo se relacionam com a ocorrência de doença coronariana em dez anos.

Para essa análise, são consideradas duas informações disponíveis no conjunto de dados: a condição de fumante atual (`currentSmoker`) e a quantidade média de cigarros consumidos por dia (`cigsPerDay`). A primeira permite comparar a ocorrência de `TenYearCHD = 1` entre fumantes e não fumantes atuais, enquanto a segunda complementa a análise com a intensidade do consumo.

In [0]:
resultado_tabagismo = (
    df_gold_validacao
    .groupBy("currentSmoker")
    .agg(
        F.count("*").alias("total_registros"),
        F.sum("TenYearCHD").alias("casos_CHD")
    )
    .withColumn(
        "percentual_CHD",
        F.round(
            (F.col("casos_CHD") / F.col("total_registros")) * 100,
            2
        )
    )
    .orderBy("currentSmoker")
)

display(resultado_tabagismo)

currentSmoker,total_registros,casos_CHD,percentual_CHD
0,2144,311,14.51
1,2094,333,15.9


Entre os indivíduos que não eram fumantes atuais, 14,51% apresentaram `TenYearCHD = 1`, enquanto entre os fumantes atuais esse percentual foi de 15,90%. A diferença entre os dois grupos foi, portanto, de 1,39 ponto percentual, indicando frequências próximas no conjunto analisado.

A quantidade de cigarros consumidos por dia, analisada na seção 5.5, complementa esse resultado. Os indivíduos com `TenYearCHD = 1` apresentaram consumo médio de 10,60 cigarros por dia, enquanto aqueles com `TenYearCHD = 0` apresentaram média de 8,64 cigarros por dia.

Assim, a condição de fumante atual apresentou pequena diferença na ocorrência de doença coronariana entre os grupos analisados. Por outro lado, a classe com ocorrência da doença apresentou maior consumo médio diário de cigarros, mostrando uma diferença mais evidente quando considerada a intensidade do consumo.

## 7.5 Diabetes e ocorrência de doença coronariana

A quinta pergunta de negócio busca verificar se pacientes com diabetes apresentam maior ocorrência de doença coronariana em dez anos quando comparados aos pacientes sem diabetes.

Para responder à questão, é comparada a proporção de registros com `TenYearCHD = 1` entre indivíduos sem diabetes (`diabetes = 0`) e com diabetes (`diabetes = 1`).

In [0]:
resultado_diabetes = (
    df_gold_validacao
    .groupBy("diabetes")
    .agg(
        F.count("*").alias("total_registros"),
        F.sum("TenYearCHD").alias("casos_CHD")
    )
    .withColumn(
        "percentual_CHD",
        F.round(
            (F.col("casos_CHD") / F.col("total_registros")) * 100,
            2
        )
    )
    .orderBy("diabetes")
)

display(resultado_diabetes)

diabetes,total_registros,casos_CHD,percentual_CHD
0,4129,604,14.63
1,109,40,36.7


Os resultados mostram que, entre os indivíduos sem diabetes, 14,63% apresentaram `TenYearCHD = 1`. Entre os indivíduos com diabetes, esse percentual foi de 36,70%, uma diferença de 22,07 pontos percentuais entre os grupos.

Portanto, no conjunto de dados analisado, a ocorrência de doença coronariana em dez anos foi mais frequente entre os pacientes com diabetes do que entre aqueles sem diabetes.

É importante considerar também a diferença no tamanho dos grupos: foram identificados 109 pacientes com diabetes e 4.129 sem diabetes. Dessa forma, os percentuais observados devem ser interpretados considerando a menor quantidade de registros disponível para o grupo com diabetes.

## 7.6 Síntese das respostas às perguntas de negócio

As análises realizadas permitiram responder às cinco perguntas de negócio definidas no início do projeto.

A ocorrência de `TenYearCHD = 1` aumentou progressivamente entre as faixas etárias, passando de 4,14% entre indivíduos de 30 a 39 anos para 27,68% entre aqueles de 60 a 70 anos. Em relação à hipertensão, a ocorrência foi de 10,92% entre indivíduos sem hipertensão e de 24,70% entre aqueles com hipertensão.

Os indicadores clínicos avaliados também apresentaram diferenças entre os grupos. Os indivíduos com `TenYearCHD = 1` apresentaram, em média, valores mais elevados de colesterol total, pressão arterial sistólica e diastólica, IMC e glicose, conforme observado na seção 7.3.

Em relação ao tabagismo, a ocorrência de `TenYearCHD = 1` foi de 14,51% entre não fumantes atuais e de 15,90% entre fumantes atuais, representando uma diferença relativamente pequena. Entretanto, os indivíduos com `TenYearCHD = 1` também apresentaram maior consumo médio diário de cigarros, conforme analisado na seção 7.4.

Por fim, entre os indivíduos sem diabetes, 14,63% apresentaram `TenYearCHD = 1`, enquanto entre aqueles com diabetes o percentual foi de 36,70%. Esse resultado deve ser interpretado considerando o menor número de indivíduos com diabetes presente no conjunto de dados.

De forma geral, as análises permitiram identificar diferenças na ocorrência de doença coronariana entre os grupos avaliados e responder às cinco perguntas de negócio propostas para o projeto.

# 8. Conclusão

Este projeto teve como objetivo construir um pipeline de dados para o conjunto Framingham, realizando etapas de ingestão, tratamento, validação, organização e análise dos dados relacionados à ocorrência de doença cardíaca coronariana no horizonte de dez anos.

A organização do pipeline em camadas Bronze, Silver e Gold permitiu preservar os dados de origem, aplicar os tratamentos necessários e disponibilizar um conjunto final estruturado para consumo analítico e etapas posteriores de modelagem. Durante esse processo, foram realizadas verificações relacionadas a valores ausentes, registros duplicados, padronização dos atributos, consistência dos valores e presença de observações extremas.

Ao final do processamento, a camada Gold foi disponibilizada com 4.238 registros e 16 atributos, sendo 15 variáveis explicativas e a variável-alvo `TenYearCHD`. A validação da tabela persistida confirmou a ausência de valores ausentes e registros duplicados, além da preservação das transformações realizadas ao longo do pipeline.

As cinco perguntas de negócio definidas no início do projeto foram respondidas a partir dos dados disponibilizados na camada Gold. As análises permitiram comparar a ocorrência de doença coronariana entre diferentes faixas etárias, pacientes com e sem hipertensão, indicadores clínicos, características relacionadas ao tabagismo e pacientes com e sem diabetes. Dessa forma, além da construção e validação do pipeline, o projeto também atingiu o objetivo de transformar os dados tratados em informações capazes de responder às questões propostas.

Entre os resultados observados, verificou-se maior frequência de `TenYearCHD = 1` nas faixas etárias mais elevadas e entre indivíduos com hipertensão ou diabetes. Os pacientes da classe positiva também apresentaram médias mais elevadas nos indicadores clínicos analisados. Em relação ao tabagismo, a diferença na ocorrência entre fumantes e não fumantes atuais foi menor, embora tenha sido observado maior consumo médio diário de cigarros entre os indivíduos com `TenYearCHD = 1`.

Também foi identificado desbalanceamento na variável-alvo, com menor representação dos registros com `TenYearCHD = 1`. Essa característica deverá ser considerada em uma eventual etapa de modelagem, principalmente na escolha das métricas de avaliação e das estratégias utilizadas durante o treinamento dos modelos.

Dessa forma, o projeto resultou em um fluxo de dados estruturado e rastreável, desde a fonte original até a camada destinada à análise. Os dados finais encontram-se tratados, validados e organizados, as questões de negócio propostas foram respondidas e o conjunto resultante está preparado para apoiar análises adicionais e futuras etapas de modelagem.